frontend/src/hooks/api/secrets/queries.tsx

```diff
@@ -147,12 +159,15 @@ export const useGetProjectSecretsAllEnv = ({
           setIsErrorHandled.on();
         }
       },
-      queryFn: async () => fetchProjectSecrets({ workspaceId, environment, secretPath }),
-      select: (el: SecretV3RawResponse) =>
-        mergePersonalSecrets(el.secrets).reduce<Record<string, SecretV3RawSanitized>>(
+      queryFn: async () => {
+        const resp = await fetchProjectSecrets({ workspaceId, environment, secretPath });
+
+        return mergePersonalSecrets(resp.secrets).reduce<Record<string, SecretV3RawSanitized>>(
           (prev, curr) => ({ ...prev, [curr.key]: curr }),
           {}
-        )
+        );
+      },
+      staleTime: 5 * 60 * 1000
```

Comment:
I increased the `staleTime` because when navigating between folders in the secrets overview, the queries would refetch everytime - causing the UI to lock up while re-fetching. This stale time allows smooth transitions and queries are invalidated on mutations so small chance of seeing outdated data